solcore/analytic_solar_cells/depletion_approximation.py

```diff
@@ -383,6 +395,52 @@ def bc(ya, yb):
     return out
 
 
+def get_J_sc_diffusion_green(xa, xb, g, D, L, y0, S, ph, side='top'):
```

Comment:
I know that `solcore` is not that great with docstrings - to put it mildly - but it will be great if any new addition were thorough with that and type annotations, to make the code more readable. Nothing to do in this first pass, but something to tkae into account.

Reply:
Docstring has been added, similar to the other functions. Do you think the newly introduced top level functions need a docstring (introduced because of your third comment)? I am not sure what you mean with type annotations.

Reply:
Ideally, yes, so the functions are really standalone. About type annotations, you have a very nice blog post describing the matter here: https://towardsdatascience.com/type-annotations-in-python-d90990b172dc